packages/hash-bundle/lite.js

```diff
@@ -0,0 +1,73 @@
+// @ts-check
+/// <reference types="ses"/>
+
+import { decodeBase64 } from '@endo/base64/decode.js';
+import { parseArchive } from '@endo/compartment-mapper/import-archive.js';
+
+const { details: d, quote: q } = assert;
+
+const textEncoder = new TextEncoder();
+
+/**
+ * @param {Record<any, any>} bundle
+ * @param {(bytes: Uint8Array) => string} computeSha512
+ * @param {string} name
+ * @returns {Promise<string>}
+ */
+export const hashBundle = async (
+  bundle,
+  computeSha512,
+  name = '<unknown-bundle>',
+) => {
+  assert(bundle !== null, d`hashBundle expects the bytes of a bundle`);
+  assert.typeof(
+    bundle,
+    'object',
+    d`Cannot hash non-bundle, must be of type object, got ${q(bundle)}`,
+  );
+  const { moduleFormat } = bundle;
+  assert.typeof(
+    moduleFormat,
+    'string',
+    d`Cannot hash non-bundle, moduleFormat must be a string, got ${typeof moduleFormat}`,
+  );
+
+  let prefix = '';
+  let sha512 = '';
+  if (moduleFormat === 'nestedEvaluate') {
+    const { source } = bundle;
+    assert.typeof(
+      source,
+      'string',
+      d`Cannot hash non-bundle, source must be a string, got ${typeof source}`,
+    );
+    const sourceBytes = textEncoder.encode(source);
+    sha512 = computeSha512(sourceBytes);
+    prefix = 'v0-';
+  } else if (moduleFormat === 'endoZipBase64') {
+    const { endoZipBase64 } = bundle;
+    assert.typeof(
+      endoZipBase64,
+      'string',
+      d`Cannot hash non-bundle, property 'endoZipBase64' must be a string, got ${typeof endoZipBase64}`,
+    );
+    const bytes = decodeBase64(endoZipBase64);
+    const { sha512: parsedSha512 } = await parseArchive(bytes, name, {
+      computeSha512,
+    });
+    assert(parsedSha512 !== undefined);
+    sha512 = parsedSha512;
+    prefix = `v1-`;
+  } else if (moduleFormat === 'getExport') {
+    assert.fail(
+      d`Cannot determine hash of bundle with getExport moduleFormat because it is not consistent across platforms`,
+    );
+  } else {
+    assert.fail(
+      d`Cannot determine hash of bundle with unrecognized moduleFormat ${q(
+        moduleFormat,
+      )}`,
+    );
+  }
+  return `${prefix}-${sha512}`;
```

Comment:
Either `prefix` should be `v1`, or the concatenation should omit the hyphen, this produces `v1--HASH`, not `v1-HASH`.